File: September, 2024/Sept_12_Leet1684/sept12.py
# Approach 1

from typing import List


class Solution:
    def countConsistentStrings(self, allowed: str, words: List[str]) -> int:
        count = 0
        for i in words:
            flag = 0
            for char in i:
                if char in allowed:
                    flag = 0
                else:
                    flag = 1
                    break
            if flag == 0:
                count = count+1
        return count 
    
# Approach 2

class Solution:
    def countConsistentStrings(self, allowed: str, words: List[str]) -> int:
        allowed_set = set(allowed)  # O(1) lookup for allowed characters
        count = 0
        
        for word in words:
            
            if all(char in allowed_set for char in word):
                count += 1
                
        return count

File: September, 2024/Sept_12_Leet1684/sept12.java
package Sept_12_Leet1684;
import java.util.Set;
import java.util.HashSet;
public class sept12 {
    public int countConsistentStrings(String allowed, String[] words) {
        Set<Character> allowed_set = new HashSet<>();

        for(char c: allowed.toCharArray()){
            allowed_set.add(c);
        }
        int count = 0;
        for(String word : words){
            boolean flag = false;
            for(char i: word.toCharArray()){
                if(allowed_set.contains(i)){
                    flag = true;
                }
                else{
                    flag = false;
                    break;
                }
            }
            if(flag){
                count++;
            }
        }

        return count;
    }
}
